src/android/Dgram.java

```diff
@@ -75,198 +48,191 @@ public void run() {
             DatagramPacket datagramPacket = new DatagramPacket(data, data.length);
             while (true) {
                 try {
-                    if (this.m_datagramSocket.isClosed()) {
+                    if (this.datagramSocket.isClosed()) {
                         Log.d(TAG, "Exiting message loop because socket is closed.");
                         return;
                     }
 
                     // Reset the length in case we receive an incomplete DatagramPacket
                     datagramPacket.setLength(data.length);
-                    this.m_datagramSocket.receive(datagramPacket);
+                    this.datagramSocket.receive(datagramPacket);
                     String message = new String(data, 0, datagramPacket.getLength(), "UTF-8");
                     String address = datagramPacket.getAddress().getHostAddress();
                     int port = datagramPacket.getPort();
                     Log.d(TAG, "Received message " + message + " from " + address);
-                    Dgram.this.m_javascriptQueue.put("_onMessage("
-                            + this.m_datagramSocketId + ","
-                            + "'" + message + "',"
-                            + "'" + address + "',"
-                            + port + ");");
-                } catch (InterruptedException e) {
-                    Log.d(TAG, e.toString());
-                } catch (Exception e) {
-                    Log.d(TAG, "Receive exception:" + e.toString());
-
-                    try {
-                        Dgram.this.m_javascriptQueue.put("_onError("
-                                + this.m_datagramSocketId
-                                + ",'receive','"
-                                + e.toString() + "');");
-                    } catch (Exception innerException) {
-                        Log.e(TAG, "Receive exception:" + innerException.toString());
-                    } finally {
-                        return;
+                    if (onMessageCallback != null) {
+                        sendMessageResult(message, address, port);
                     }
+                } catch (Exception e) {
+                    Log.d(TAG, "Received exception:" + e.toString());
+                    sendMessageErrorResult(e);
                 }
             }
         }
     }
 
+    private void sendMessageResult(String message, String address, int port) {
```

Comment:
i would prefer reserving `send` as a verb for pushing a message out on the socket to another device and give communicating information on the callback another verb.  some thoughts are `on`, `emit`, `push`, `notify`, `communicate`.  for example, if i saw `onMessageResult` i would assume it was a handler for callback.  i also like `emitMessageResult` to clarify that it is pushing the message from the cordova layer to javascript.